pkgs/development/libraries/udns/default.nix

```
@@ -16,7 +16,20 @@ stdenv.mkDerivation rec {
     sha256 = "0447fv1hmb44nnchdn6p5pd9b44x8p5jn0ahw6crwbqsg7f0hl8i";
   };
 
+  # udns uses a very custom build and hardcodes a .so name in a few places.
+  # Instead of fighting with it to apply the standard dylib script, change
+  # the right place in the Makefile itself.
+  postPatch =
+    if stdenv.isDarwin
+    then
+      ''
+        substituteInPlace Makefile.in \
+          --replace --soname, -install_name,$out/lib/
+      ''
+    else "";
```

quotes are not needed but improve readability here.

```suggestion
  postPatch = lib.optionalString stdenv.isDarwin ''
    substituteInPlace Makefile.in \
      --replace "--soname," "-install_name,$out/lib/"
  '';
```